### PR TITLE
Implement of the Json codec to use a circe

### PR DESCRIPTION
**Pull request by tkrs**

### CODE REVIEW

File: core/src/main/scala/aerospiker/AsyncWrite.scala [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    logger.debug(key.toString)
+    val fields = Seq(
+      writeField(key.namespace, FieldType.NAMESPACE, 0, true),
+      writeField(key.setName, FieldType.TABLE, 0, true),
+      writeField(key.digest, FieldType.DIGEST_RIPE, 0, true),
+      writeField(key.userKey.toString, FieldType.KEY, 1, policy.sendKey)
+    ) collect {
+        case Some(a) => a
+      }
```
Comment (yanana) [inline core/src/main/scala/aerospiker/AsyncWrite.scala]: Incorrect indentation

File: msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  @throws(classOf[AerospikeException])
+  def unpackObjectList(buffer: Array[Byte]): Json =
+    new JsonUnpacker(buffer, 0, buffer.length).unpackList
+
+  @throws(classOf[AerospikeException])
+  def unpackObjectMap(buffer: Array[Byte]): Json =
+    new JsonUnpacker(buffer, 0, buffer.length).unpackMap
+}
+
+class JsonUnpacker(buffer: Array[Byte], _offset: Int, length: Int) {
```
Comment (yanana) [inline msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala]: Use `class JsonUnpacker(buffer: Array[Byte], var offset: Int, length: Int) {` instead.

File: msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+import java.io.IOException
+import com.aerospike.client.AerospikeException
+import com.aerospike.client.command.Buffer
+import com.aerospike.client.command.ParticleType
+
+import io.circe._
+import scala.collection.mutable.ListBuffer
+
+// TODO: Improve implement
```
Comment (yanana) [inline msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala]: s/implement/implementation

File: task/src/main/scala/aerospiker/task/Operation.scala [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+
+  def get[R](a: String)(
+    implicit
+    decoder: Decoder[Map[String, R]]
+  ): Task[NoSuchKey Xor Option[R]] = {
+    val t = Task.async[Option[R]] { register =>
+      command.execute[Seq[String], Map[String, R]](
+        Some(new ExecuteListener[Map[String, R]] {
+          override def onFailure(e: AerospikeException): Unit = {
+            e.printStackTrace()
```
Comment (yanana) [inline task/src/main/scala/aerospiker/task/Operation.scala]: Why not use logger?

File: msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala (hunk 2 above)
Comment (tkrs) [inline msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala]: Exactly!

File: task/src/main/scala/aerospiker/task/Operation.scala (hunk 4 above)
Comment (tkrs) [inline task/src/main/scala/aerospiker/task/Operation.scala]: oh, sorry! I will remove it.

File: msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala (hunk 3 above)
Comment (tkrs) [inline msgpack/src/main/scala/aerospiker/msgpack/JsonUnpacker.scala]: I see

File: core/src/main/scala/aerospiker/AsyncWrite.scala (hunk 1 above)
Comment (tkrs) [inline core/src/main/scala/aerospiker/AsyncWrite.scala]: I know.. but,  it is the applied `scalariform` .

**Pull request merged**